server-common/src/main/java/org/apache/kafka/server/group/share/PersisterStateBatch.java

```diff
@@ -95,4 +95,18 @@ public String toString() {
             "deliveryCount=" + deliveryCount +
             ")";
     }
+
+    @Override
+    public int compareTo(Object o) {
+        PersisterStateBatch that = (PersisterStateBatch) o;
+        int deltaFirst = Long.compare(this.firstOffset(), that.firstOffset());
+        if (deltaFirst == 0) {
+            int deltaLast = Long.compare(this.lastOffset(), that.lastOffset());
+            if (deltaLast == 0) {
+                return this.deliveryCount() - that.deliveryCount();
+            }
+            return deltaLast;
+        }
+        return deltaFirst;
+    }
```

Comment:
What about `deliveryState` is that not important to compare for this class? If not then can we please write a comment regarding why it's skipped.

Reply:
its managed in the next revision